packages/ipfs-core/src/components/add-all/index.js

```diff
@@ -58,8 +58,7 @@ module.exports = ({ block, gcLock, preload, pin, options: constructorOptions })
   /**
    * @type {AddAll<{}>}
    */
-  async function * addAll (source, options) {
-    options = options || {}
+  async function * addAll (source, options = {}) {
```

Comment:
@achingbrain I can not because if we don't provide the default type checker assigns  it a `undefined|Options` type. And that type remains even after assignment `options = options || {}` (because types do not mutate). So if we don't want to provide default then we have to introduce another variable to deal with that e.g.:

```
const options2 = options || {}
```

Which will then have `Options` type instead of `Options|void`. The reason I did not choose later approach, because it ends up creating a lot more changes then this.

@achingbrain given this information, could you please let me know what should I do here and all other places like this ?

Reply:
> For consistency each core API method should have this change applied.

Do you want this part of this pull request or a followup ?